proxy/proxy.go

```diff
@@ -12,27 +12,33 @@ import (
 
 // MakeHTTPClientWithDisableKeepAlives makes a HTTP client with good defaults for timeouts.
 func MakeHTTPClientWithDisableKeepAlives(timeout *time.Duration, tlsInsecure bool, disableKeepAlives *bool) http.Client {
-	if timeout != nil {
-		tr := &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
```

Comment:
Ed, this will break the CLI for our enterprise users. Why was it removed?

Reply:
not on perpose, mistake, added back

Reply:
@alexellis just occurred to me that proxy is only set when timeout is not nil. Is this the desired behaviour? What is going to happen when an enterprise user runs `faas-cli invoke`?

Reply:
No, the proxy needs to be there all the time.

Reply:
```
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
```

Seems like (you also pointed out) - the proxy is set by default in the `DefaultTransport`.

Reply:
yep i've pushed a new commit with that change in it